internal/glance/glance.go

```diff
@@ -439,6 +442,23 @@ func (a *application) server() (func() error, func() error) {
 	mux.HandleFunc("GET /{$}", a.handlePageRequest)
 	mux.HandleFunc("GET /{page}", a.handlePageRequest)
 
+	var absDataPath string
+	if a.Config.Server.DataPath == "" {
+		a.Config.Server.DataPath = "./data"
+	}
+	absDataPath, _ = filepath.Abs(a.Config.Server.DataPath)
+
+	fs := afero.NewOsFs()
+	todosDataPath := path.Join(absDataPath, "todos")
+	exist, _ := afero.DirExists(fs, todosDataPath)
+	if !exist {
+		fs.MkdirAll(todosDataPath, os.ModePerm)
+	}
+
+	todoWidget := &todoWidget{storage: afero.NewBasePathFs(fs, todosDataPath)}
+	mux.HandleFunc("GET /api/widgets/todo/{id}", todoWidget.handleRequest)
+	mux.HandleFunc("PUT /api/widgets/todo/{id}", todoWidget.handleRequest)
```

Comment:
Doing this here creates an explicit binding to the todo component. It doesn't allow the application to be independent of a single component (todo in this case). I'd like to suggest moving this logic into the widget itself, maybe defined through an interface to standardize this procedure for future usage in other/new components.